presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -159,6 +161,36 @@ protected PlanWithProperties visitPlan(PlanNode node, PreferredProperties prefer
         }
 
         @Override
+        public PlanWithProperties visitProject(ProjectNode node, PreferredProperties preferred)
+        {
+            Map<Symbol, Symbol> identities = computeIdentityTranslations(node.getAssignments());
+            List<LocalProperty<Symbol>> localProperties = LocalProperties.translate(preferred.getLocalProperties(), column -> Optional.ofNullable(identities.get(column)));
+
+            if (preferred.getPartitioningProperties().isPresent()) {
```

Comment:
All this logic would seem to belong in `determineChildPreferences`. It may be possible to do it by simply extending it to take the translation map as an argument and applying the transformations as needed.